test/utils/InstanceTest.sol

```diff
@@ -315,12 +315,59 @@ abstract contract InstanceTest is HyperdriveTest {
 
     /// @dev A virtual function that converts an amount in terms of the base token
     ///      to equivalent amount in shares.
-    /// @param baseAmount Amount in terms of the base.
+    /// @param baseAmount Amount in terms of base.
     /// @return shareAmount Amount in terms of shares.
     function convertToShares(
         uint256 baseAmount
     ) internal view virtual returns (uint256 shareAmount);
 
+    /// @dev A virtual function that converts an amount in terms of the share token
+    ///      to equivalent amount in base.
+    /// @param shareAmount Amount in terms of shares.
+    /// @return baseAmount Amount in terms of base.
+    function convertToBase(
+        uint256 shareAmount
+    ) internal view virtual returns (uint256 baseAmount);
+
+    /// @dev A virtual function that ensures the deposit accounting is correct
+    ///      when opening positions.
+    /// @param trader The account opening the position.
+    /// @param basePaid The amount the position was opened with in terms of base.
+    /// @param asBase Flag to determine whether the position was opened with the base or share token.
+    /// @param totalBaseBefore Total supply of the base token before the trade.
+    /// @param totalSharesBefore Total supply of the share token before the trade.
+    /// @param traderBalancesBefore Balances of tokens of the trader before the trade.
+    /// @param hyperdriveBalancesBefore Balances of tokens of the Hyperdrive contract before the trade.
+    function verifyDeposit(
+        address trader,
+        uint256 basePaid,
+        bool asBase,
+        uint256 totalBaseBefore,
+        uint256 totalSharesBefore,
+        AccountBalances memory traderBalancesBefore,
+        AccountBalances memory hyperdriveBalancesBefore
+    ) internal virtual;
+
+    /// @dev A virtual function that fetches the token balance information of an account.
+    /// @param account The account to fetch token balances of.
+    /// @return sharesBalance The shares token balance of the account.
+    /// @return baseBalance The base token balance of the account.
+    function getTokenBalances(
+        address account
+    )
+        internal
+        view
+        virtual
+        returns (uint256 sharesBalance, uint256 baseBalance);
```

Comment:
Should this be base and then shares? That would be more consistent with the below function.